lib/java_buildpack/jre/ibmjre_like.rb

```diff
@@ -0,0 +1,174 @@
+# Encoding: utf-8
+
+# Cloud Foundry Java Buildpack
+# Copyright 2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'fileutils'
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/jre'
+require 'java_buildpack/util/tokenized_version'
+
+module JavaBuildpack
+  module Jre
+
+    # Encapsulates the detect, compile, and release functionality for selecting a JRE.
+    class IbmjreLike < JavaBuildpack::Component::VersionedDependencyComponent
```

Comment:
Unless there is a plan for more than one IBM JRE-like implementation, there's no reason to name this "-like".